aws/resource_aws_s3_bucket_inventory.go

```diff
@@ -0,0 +1,445 @@
+		filterMap := filterList[0].(map[string]interface{})
+		inventoryConfiguration.Filter = expandS3InventoryFilter(filterMap)
+	}
+
+	if v, ok := d.GetOk("destination"); ok {
+		destinationList := v.([]interface{})
+		destinationMap := destinationList[0].(map[string]interface{})
+		bucketList := destinationMap["bucket"].([]interface{})
+		bucketMap := bucketList[0].(map[string]interface{})
+
+		inventoryConfiguration.Destination = &s3.InventoryDestination{
+			S3BucketDestination: expandS3InventoryS3BucketDestination(bucketMap),
+		}
+	}
+
+	input := &s3.PutBucketInventoryConfigurationInput{
+		Bucket: aws.String(bucket),
+		Id:     aws.String(name),
+		InventoryConfiguration: inventoryConfiguration,
+	}
+
+	log.Printf("[DEBUG] Putting S3 bucket inventory configuration: %s", input)
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := conn.PutBucketInventoryConfiguration(input)
+		if err != nil {
+			if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Error putting S3 bucket inventory configuration: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", bucket, name))
+
+	return resourceAwsS3BucketInventoryRead(d, meta)
+}
+
+func resourceAwsS3BucketInventoryDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).s3conn
+
+	bucket, name, err := resourceAwsS3BucketInventoryParseID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	input := &s3.DeleteBucketInventoryConfigurationInput{
+		Bucket: aws.String(bucket),
+		Id:     aws.String(name),
+	}
+
+	log.Printf("[DEBUG] Deleting S3 bucket inventory configuration: %s", input)
+	_, err = conn.DeleteBucketInventoryConfiguration(input)
+	if err != nil {
+		if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") || isAWSErr(err, "NoSuchConfiguration", "The specified configuration does not exist.") {
+			return nil
+		}
+		return fmt.Errorf("Error deleting S3 bucket inventory configuration: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsS3BucketInventoryRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).s3conn
+
+	bucket, name, err := resourceAwsS3BucketInventoryParseID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("bucket", bucket)
+	d.Set("name", name)
+
+	input := &s3.GetBucketInventoryConfigurationInput{
+		Bucket: aws.String(bucket),
+		Id:     aws.String(name),
+	}
+
+	log.Printf("[DEBUG] Reading S3 bucket inventory configuration: %s", input)
+	output, err := conn.GetBucketInventoryConfiguration(input)
+	if err != nil {
+		if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") || isAWSErr(err, "NoSuchConfiguration", "The specified configuration does not exist.") {
+			log.Printf("[WARN] %s S3 bucket inventory configuration not found, removing from state.", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("enabled", *output.InventoryConfiguration.IsEnabled)
```

Comment:
To prevent potential panics, we should prefer to use the SDK provided function for dereferencing values (`aws.BoolValue(output.InventoryConfiguration.IsEnabled)`) instead of directly dereferencing via `*`.

Same applies to `included_object_versions` below.